planner/core/planbuilder.go

```diff
@@ -2130,12 +2130,14 @@ func (b *PlanBuilder) buildDDL(ctx context.Context, node ast.DDLNode) (Plan, err
 			return nil, err
 		}
 		schema := plan.Schema()
-		if v.Cols != nil && len(v.Cols) != schema.Len() {
-			return nil, ddl.ErrViewWrongList
+		if v.Cols == nil {
```

Comment:
If we run `create view v(a,b,c) as selet b,d,e from t`.
v.Cols here is [a,b,c] ?

Reply:
yes, we just need to colname to build projection.